Tests/ClientRuntimeTests/InterceptorTests/InterceptorTests.swift

```diff
@@ -0,0 +1,119 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import XCTest
+@testable import ClientRuntime
+
+class InterceptorTests: XCTestCase {
+    struct TestInput {
+        public var property: String?
+        public var otherProperty: Int = 0
+    }
+
+    struct TestOutput {
+        public var property: String?
+    }
+
+    struct AddAttributeInterceptor<T, RequestType, ResponseType, AttributesType: HasAttributes> : Interceptor {
+        private let key: AttributeKey<T>
+        private let value: T
+
+        init(key: AttributeKey<T>, value: T) {
+            self.key = key
+            self.value = value
+        }
+
+        public func modifyBeforeSerialization(context: inout some MutableInput<Self.AttributesType>) async throws {
+            let attributes = context.getAttributes()
+            attributes.set(key: self.key, value: self.value)
+        }
+    }
+
+    struct ModifyInputInterceptor<InputType, RequestType, ResponseType, AttributesType: HasAttributes> : Interceptor {
+        private let keyPath: WritableKeyPath<InputType, String?>
+        private let value: String
+
+        init(keyPath: WritableKeyPath<InputType, String?>, value: String) {
+            self.keyPath = keyPath
+            self.value = value
+        }
+
+        public func modifyBeforeSerialization(context: inout some MutableInput<Self.AttributesType>) async throws {
+            var input: InputType = context.getInput()!
+            input[keyPath: keyPath] = value
+            context.updateInput(updated: input)
+        }
+    }
+
+    struct AddHeaderInterceptor : HttpInterceptor {
+        private let headerName: String
+        private let headerValue: String
+
+        init(headerName: String, headerValue: String) {
+            self.headerName = headerName
+            self.headerValue = headerValue
+        }
+
+        public func modifyBeforeTransmit(context: inout some MutableRequest<Self.RequestType, Self.AttributesType>) async throws {
+            let builder = context.getRequest().toBuilder()
+            builder.withHeader(name: headerName, value: headerValue)
+            context.updateRequest(updated: builder.build())
+        }
+    }
+
+    struct ModifyMultipleInterceptor: HttpInterceptor {
+        private let newInputValue: Int
+
+        init(newInputValue: Int) {
+            self.newInputValue = newInputValue
+        }
+
+        public func modifyBeforeSerialization(context: inout some MutableInput<Self.AttributesType>) async throws {
+            var input: TestInput = context.getInput()!
+            input.otherProperty = newInputValue
+            context.updateInput(updated: input)
+        }
+
+        public func modifyBeforeTransmit(context: inout some MutableRequest<Self.RequestType, Self.AttributesType>) async throws {
+            let input: TestInput = context.getInput()!
```

Comment:
Force unwrapping during a test will crash the entire test suite if the unwrapped value is nil.

Instead, use `XCTUnwrap` which will throw a Swift error and just fail the test being run:
```swift
let input: TestInput = try XCTUnwrap(context.getInput())
```